projects/packages/search/src/initializers/class-jetpack-initializer.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Search initializer for the Jetpack plugin.
+ *
+ * @package    @automattic/jetpack-search
+ */
+
+namespace Automattic\Jetpack\Search;
+
+/**
+ * Initializer for the main Jetpack plugin.
+ * Instantiate this initializer to enable Jetpack Search functionality.
+ */
+class Jetpack_Initializer extends Initializer {
+	/**
+	 * Initializes either the Classic Search or the Instant Search experience.
+	 */
+	public static function initialize() {
+		// Check whether Jetpack Search should be initialized in the first place.
+		if ( ! self::is_connected() || ! self::is_search_supported() ) {
+			/**
+			 * Fires when the Jetpack Search fails and would fallback to MySQL.
+			 *
+			 * @since Jetpack 7.9.0
+			 * @param string $reason Reason for Search fallback.
+			 * @param mixed  $data   Data associated with the request, such as attempted search parameters.
+			 */
+			do_action( 'jetpack_search_abort', 'inactive', null );
+			return;
+		}
+
+		$blog_id = \Jetpack::get_option( 'id' );
+		if ( ! $blog_id ) {
+			do_action( 'jetpack_search_abort', 'no_blog_id', null );
+			return;
+		}
+
+		// TODO: Port the search widget to package for milestone 2.
+		require_once JETPACK__PLUGIN_DIR . 'modules/widgets/search.php';
+
+		/**
+		 * Location of built instant search assets, does not include trailing slash.
+		 */
+		define( 'JETPACK_SEARCH_PLUGIN_DIRECTORY', dirname( dirname( __DIR__ ) ) );
+
+		if ( Options::is_instant_enabled() ) {
```

Comment:
The logic here seems common to me, which means it should be in the parent class?

Reply:
Perhaps? I'll reorganize the class structure here once we have another initializer to find commonalities with.

Reply:
I wonder whether the initializer should reside in Jetpack the plugin, with only the shared logic in the search package.

Reply:
Definitely worth exploring in the future, given that we likely won't commit `Wpcom_Initializer` here in the Jetpack repo.